src/main/kotlin/org/rust/ide/utils/RustModTabTitleProvider.kt

```diff
@@ -0,0 +1,18 @@
+package org.rust.ide.utils
+
+import com.intellij.openapi.fileEditor.impl.EditorTabTitleProvider
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import org.rust.lang.core.psi.impl.isRustFile
+
+class RustModTabTitleProvider : EditorTabTitleProvider {
+    val MOD_FILE_NAME = "mod.rs"
+
+    override fun getEditorTabTitle(project: Project, file: VirtualFile): String? {
+        if (file.isRustFile && MOD_FILE_NAME == file.name) {
+            return "${file.parent.name}/${file.name}"
```

Comment:
This presentation may be still unambiguous for following situation (I tend to use such module structure, so this is real life example):

- /foo
  - /moo
  - /macros
    - **/mod.rs** => `macros/mod.rs`
    - /macro_a.rs
    - ...
- /bar
  - /boo
  - /macros
    - **/mod.rs** => `macros/mod.rs`
    - /macro_b.rs
    - ...

Though I don't know yet how to handle this. Showing module name may be ugly while trying to append parent dirs until there are no ambiguities can be overworking simple problem.